lifecycle.Rmd

```diff
@@ -7,52 +7,251 @@ status("drafting")
+A change that is not backward compatible is often described as a *breaking* change.
+In section @sec-lifecycle-backward-compatibility, we'll revisit the topic of backward compatibility.
+Similar to the issue of package dependencies (@sec-dependencies-pros-cons), it's a question of trade-offs.
+
+In practice, backward compatibility is not a clear-cut distinction.
+It is typical to assess the impact of a change from a few angles:
+
+-   Degree of change in behaviour.
+    The most extreme is to make something that used to be possible into an error, i.e. impossible.
+    In contrast, changes to a print method are much less significant.
+
+-   How the changes fit into the design of the package.
+    A change to low-level infrastructure, such as a utility that gets called in all user-facing functions, is more fraught than a change that only affects one parameter of a single function.
+
+-   How much existing usage is affected.
+    This is a combination of how many of your users will perceive the change and how many existing users there are to begin with.
+
+Recall that a version number will have one of these forms, if you're following our conventions:
+
+``` default
+<major>.<minor>.<patch>        # released version
+<major>.<minor>.<patch>.<dev>  # in-development version
+```
+
+If the current package version is `0.8.1.9000`, here's our advice on how to pick the version number for the next release:
+
+-   Increment `patch`, e.g. `0.8.2` for a **patch release**: you've fixed bugs, but you haven't added any significant new features.
+    For example, if we discover a show-stopping bug shortly after a release, we would make a quick patch release with the fix.
+    Most releases will have a patch number of 0.
+
+-   Increment `minor`, e.g. `0.9.0`, for a **minor release**.
+    A minor release can include bug fixes, new features, and changes that are backward compatible[^lifecycle-4].
+    This is the most common type of release.
+    It's perfectly fine to have so many minor releases that you need to use two (or even three!) digits, e.g. `1.17.0`.
+
+-   Increment `major`, e.g. `1.0.0`, for a **major release**.
+    This is best reserved for changes that are not backward compatible and that are likely to affect many users.
+    The `1.0.0` release has special significance and typically indicates that your package is feature complete with a stable API.
 
-```{r, eval = FALSE}
+[^lifecycle-4]: For some suitably pragmatic definition of "backward compatible".
+
+The trickiest decision you are likely to face is whether a change is "breaking" enough to deserve a major release.
+For example, if you make an API-incompatible change to a rarely-used part of your code, it may not make sense to increase the major number.
+But if you fix a bug that many people depend on (it happens!), it will feel like a breaking change to those folks.
+It's conceivable that such a bug fix could merit a major release.
```

Comment:
I think it's worth mentioning something about marketing here — people are more likely to pay attention to major releases and you want to take advantage of that.

Reply:
Added